ext/opcache/zend_accelerator_util_funcs.c

```diff
@@ -156,6 +157,9 @@ static void zend_accel_function_hash_copy(HashTable *target, HashTable *source)
 			goto failure;
 		}
 		_zend_hash_append_ptr_ex(target, p->key, Z_PTR(p->val), 1);
+		if (*ZSTR_VAL(p->key)) { // if not rtd key
+			zend_observer_function_declared_notify(Z_PTR(p->val), p->key);
+		}
```

Comment:
This loop is really hot. Adding a call will cause a performance drop on each request.

Reply:
I suppose we can expose in zend_observer.h whether we have any registered observers for this, eventually short-circuiting this and getting perfect branch prediction here?